### PR TITLE
New version: Twingate.Client version 2024.99.6390

### PR DESCRIPTION
**Pull request by chenbishop**
Checklist for Pull Requests
- [ ] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [ ] Is there a linked Issue?

Manifests
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/doc/Authoring.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.6 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.6.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

---

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/152852)

### CODE REVIEW

**Comment by chenbishop** [comment]
Tested with 

winget validate --manifest <path>
winget install --manifest <path>

and seems to be working locally. Any recommandations?

**Comment by stephengillie** [comment]
Automatic Validation ended with:

> Installation failed with exit code -1978335226
> 2024-05-09 07:30:06.510 [CLI ] ShellExecute installer failed: 1603
> get_hostfxr_path failed: '-2147450749'
> CustomAction CheckNetCoreRuntime returned actual error code 1603 but will be translated to success due to continue marking
> Property(S): ErrorDialog = ErrorDlg
> MSI (s) (50:2C) [07:30:06:359]: Product: Twingate 2024.99.6390 -- Installation failed.

Automated error analysis suggests 1603 may mean Install successful but system restart may be required.

Automated error analysis suggests -1978335226 may mean Install successful but system restart may be required.

(Automated response - build 880.)

**Comment by chenbishop** [comment]
Hi @stephengillie ,

As mentioned previously, I tested with

winget validate --manifest
winget install --manifest

and seems to be working locally. Could you recommend next step please?

**Comment by stephengillie** [comment]
Manual Validation ended with:
> 11:41:05 AM: Product: Twingate 
> 2024.99.6390 -- .NET Desktop Runtime 6.0 needs to be installed for this installation to continue.

(Automated response - build 880.)

File: manifests/t/Twingate/Client/2024.99.6390/Twingate.Client.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.6.0.schema.json
+
+PackageIdentifier: Twingate.Client
+PackageVersion: 2024.99.6390
+InstallerLocale: en-US
+InstallerType: msi
+InstallerSwitches:
+  Silent: /qn
+ProductCode: '{11BA3B45-D8FA-4524-A085-80FB4B023483}'
+Installers:
```
Comment (stephengillie) [inline manifests/t/Twingate/Client/2024.99.6390/Twingate.Client.installer.yaml:11]:
```suggestion
Dependencies:
  PackageDependencies:
     - PackageIdentifier: Microsoft.DotNet.DesktopRuntime.6
Installers:
```

Comment (stephengillie) [reply manifests/t/Twingate/Client/2024.99.6390/Twingate.Client.installer.yaml:11]: With the dependency, it instyalls and launches normally. 
image

**Comment by stephengillie** [comment]
Hi @chenbishop,

> This PR's version number 2024.99.6390 has 3 parameters (sets of numbers between dots - major, minor, etc), which is less than the current manifest's version 1.0.33.24402, which has 4 parameters.

Does this PR's version number **exactly** match the version reported in the Apps & features Settings page? (Feel free to attach a screenshot.)

(Automated response - build 880)

**Comment by chenbishop** [comment]
@stephengillie

The version format for Twingate has changed, see attached screenshot.

image

**Comment by stephengillie** [comment]
Hi @chenbishop,

> This PR's version number 2024.99.6390 has 3 parameters (sets of numbers between dots - major, minor, etc), which is less than the current manifest's version 1.0.33.24402, which has 4 parameters.

Does this PR's version number **exactly** match the version reported in the Apps & features Settings page? (Feel free to attach a screenshot.)

(Automated response - build 880)

File: manifests/t/Twingate/Client/2024.99.6390/Twingate.Client.installer.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# Created using wingetcreate 1.6.1.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.6.0.schema.json
+
+PackageIdentifier: Twingate.Client
+PackageVersion: 2024.99.6390
+InstallerLocale: en-US
+InstallerType: msi
+InstallerSwitches:
+  Silent: /qn
+ProductCode: '{11BA3B45-D8FA-4524-A085-80FB4B023483}'
```
Comment (stephengillie) [inline manifests/t/Twingate/Client/2024.99.6390/Twingate.Client.installer.yaml:10]:
```suggestion
ProductCode: '{11BA3B45-D8FA-4524-A085-80FB4B023483}'
AppsAndFeaturesEntries:
- DisplayVersion: 20.24.99.6390
```

**Comment by stephengillie** [comment]
Hi @chenbishop,

This manifest adds Apps and Features entries that aren't present in previous PR versions. These entries should be added to the previous versions, or removed from this version.

(Automated response - build 880.)

**Comment by chenbishop** [comment]
@stephengillie Thank you very much. I have updated the display version to 2024.99.6390, the same value as PackageVersion.

**Comment by stephengillie** [comment]
> @ stephengillie Thank you very much. Yes the version schema has been changed. I have updated the display version to 2024.99.6390, the same value as PackageVersion. Are you suggesting we should be remove all the manifest of the previous version?

Yes, it's usually a good idea to remove the previous schema versions. I believe that `2024` would be sorted higher than `1`, so it might not be necessary in this case. But would be necessary if the situation were reversed.

**Review by stephengillie — APPROVED** [review]

**Pull request merged**